src/datasources/network/network.service.interface.ts

```diff
@@ -4,19 +4,16 @@ import { NetworkResponse } from '@/datasources/network/entities/network.response
 export const NetworkService = Symbol('INetworkService');
 
 export interface INetworkService {
-  get<T = any, R = NetworkResponse<T>>(
+  get<T>(
     url: string,
     networkRequest?: NetworkRequest,
-  ): Promise<R>;
+  ): Promise<NetworkResponse<T>>;
 
-  post<T = any, R = NetworkResponse<T>>(
+  post<T>(
     url: string,
     data: object,
     networkRequest?: NetworkRequest,
-  ): Promise<R>;
+  ): Promise<NetworkResponse<T>>;
 
-  delete<T = any, R = NetworkResponse<T>>(
-    url: string,
-    data?: object,
-  ): Promise<R>;
+  delete<T>(url: string, data?: object): Promise<NetworkResponse<T>>;
```

Comment:
These types were not correct. They were previously returning `any`. Changing these caused the other type fixes in this PR.

Reply:
`any` was the default right? So a return type could hint on what value `T` should've taken here (talking about the changes before this diff).

Reply:
Very true. I've reverted the default value removal in c9ba85577a8d93c61ebf5d8aa38d80fbcaf2aebe, as well as removed the associated generics that were required because of the change.

Reply:
We can discuss the value of this default too. I was not necessarily against the change proposed here but wondering that the type could come from somewhere else (even if not explicitly declared in the generic) 😄